### PR TITLE
test(ui): create renderWithTheme function for tests (#123)

### LINKED ISSUE #123: Crete custom render function for tests
We want to create a custom `renderWithTheme` function that has the same interface as the default `render` from React Testing Library but wraps provided component in ThemeProvider

### PR DESCRIPTION
**Pull request by hoolek77**
Closes #123

### CODE REVIEW

File: packages/ui/src/utils/renderWithTheme.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { ReactElement } from 'react';
```
Comment (KonradSzwarc) [inline packages/ui/src/utils/renderWithTheme.ts:1]: 💭 For consistency with [another PR](https://github.com/CodersCrew/coderscamp/pull/126) add this function to `testHelpers.ts` file in the `src` directory

File: packages/ui/src/utils/renderWithTheme.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { render, RenderOptions, RenderResult } from '@testing-library/react';
+
+import { ThemeProvider } from '../theme';
+
+export const renderWithTheme = (ui: ReactElement, options: Omit<RenderOptions, 'queries'> = {}): RenderResult => {
```
Comment (KonradSzwarc) [inline packages/ui/src/utils/renderWithTheme.ts:7]: 🤔 This function should have the same interface the default `render` has. Here you omit the `queries` property and disallow to provide additional wrappers (they would overwrite the `ThemeProvider` if passed)

**Review by KonradSzwarc — CHANGES_REQUESTED** [review]

File: packages/ui/src/testHelpers.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+export const renderWithTheme = <
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  Q extends Queries = typeof queries,
+  Container extends Element | DocumentFragment = HTMLElement,
+>(
+  ui: React.ReactElement,
+  options: RenderOptions<Q, Container> = {},
+): RenderResult<Q, Container> => {
+  return render(<ThemeProvider>{ui}</ThemeProvider>, options);
+};
```
Comment (KonradSzwarc) [inline packages/ui/src/testHelpers.tsx:32]: 💭 As `render` is a function with type overload you can preserve it by writing code like the following:
```
const withTheme = <Render extends typeof render>(renderFc: Render) =>
  ((...args: Parameters<Render>) => {
    const [ui, options] = args;

    return renderFc(<ThemeProvider>{ui}</ThemeProvider>, options);
  }) as Render;

export const renderWithTheme = withTheme(render);
```
This way you also don't rely explicitly on current typings of separate render params and return types.

File: packages/ui/src/testHelpers.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-console */
```
Comment (KonradSzwarc) [inline packages/ui/src/testHelpers.tsx:1]: 🤔 Why do we need this?

File: packages/ui/src/testHelpers.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (KonradSzwarc) [inline packages/ui/src/testHelpers.tsx:4]: 💭 You can add `'**/testHelpers.tsx'` to `FILES_WITH_DEV_DEPENDENCIES` array in the `.eslintrc.js`

**Review by KonradSzwarc — CHANGES_REQUESTED** [review]

File: packages/ui/src/testHelpers.tsx (hunk 4 above)
Comment (hoolek77) [reply packages/ui/src/testHelpers.tsx:1]: This one comes from the other PR that implemented `withExpectedError`

**Review by KonradSzwarc — APPROVED** [review]

**Review by KonradSzwarc — APPROVED** [review]

**Review by KrystianKjjk — APPROVED** [review]

**Pull request merged**